src/foundry/common/data/fields.mjs.d.ts

```diff
@@ -420,6 +437,20 @@ interface EmbeddedCollectionField<
   isCollection: true;
 }
 
+/**
+ * A special field which contains a data object defined from the game System model.
+ * @param document - The Document class definition
+ */
+export declare function systemDataField<
+  DocumentSpecifier extends { readonly documentName: keyof Game.SystemData<any>['model'] }
+>(document: DocumentSpecifier): SystemDataField;
+// TODO: Improve
+interface SystemDataField extends DocumentField<any> {
+  type: typeof Object;
+  default: (data: { type?: string }) => Partial<Record<string, Partial<Record<string, unknown>>>> | {};
```

Comment:
I think we can leave out the `Partial`s. And `{}` is also not really needed, that case is already covered by the `Record`.